src/Stack/Types/Build.hs

```diff
@@ -629,14 +629,36 @@ configureOptsNoDir econfig bco deps isLocal package = concat
                        flagNameString name)
                     (Map.toList flags)
     , map T.unpack $ packageCabalConfigOpts package
-    , concatMap (\x -> [compilerOptionsCabalFlag wc, T.unpack x]) (packageGhcOptions package)
+    , processGhcOptions (packageGhcOptions package)
     , map ("--extra-include-dirs=" ++) (configExtraIncludeDirs config)
     , map ("--extra-lib-dirs=" ++) (configExtraLibDirs config)
     , maybe [] (\customGcc -> ["--with-gcc=" ++ toFilePath customGcc]) (configOverrideGccPath config)
     , ["--exact-configuration"]
     , ["--ghc-option=-fhide-source-paths" | hideSourcePaths cv]
     ]
   where
+    processGhcOptions :: [Text] -> [String]
+    processGhcOptions ("+RTS" : xs) =
+        let
+            (rtsArgs, rest) =
+                takeRtsArgs xs
+        in
+            ("--ghc-options=+RTS " ++ rtsArgs) : processGhcOptions rest
+    processGhcOptions (x : xs) =
+        [compilerOptionsCabalFlag wc, T.unpack x] ++ processGhcOptions xs
+    processGhcOptions [] =
+        []
+    takeRtsArgs :: [Text] -> (String, [Text])
+    takeRtsArgs ("-RTS" : xs) =
+        ("-RTS", xs)
+    takeRtsArgs (x : xs) =
+        let
+            (other, rest) =
+                takeRtsArgs xs
+        in
+            (T.unpack x ++ " " ++ other, rest)
```

Comment:
I'm not sure how important this is in this case, but this has quadratic performance. I'd recommend instead collecting a list of `Text`s and then combining with `T.unwords`. In fact, I think then you'd be able to use `break (== "-RTS")`

Reply:
Thanks for the tip! I've provided a much nicer looking implementation, and it even made me aware of the case where a user *does not* provide a `-RTS` in the argument list.

I decided to splice a `-RTS` on the end of the collected RTS args regardless. This matches the syntax you might use in a CLI invocation - `ghc --version +RTS -N -A128m`. As far as I know, this shouldn't cause any problems - any string without an `-RTS` would completely fail without this, and any string with non-RTS arguments after a `+RTS` would fail in either case.